Pandas 1.2.1 causing tests to fail

When putting a PR together I noticed that some tests were failing that were not failing before. Luckily the PR was about better dependency transparency! Looking at the diff between conda envs used between the successful workflow and the now failing one, I saw:

```
85c85
< pandas=1.2.1=py38h4c96930_0
---
> pandas=1.2.0=py38h4c96930_1
```

Tests failing:
```
=========================== short test summary info ===========================
FAILED tests/test_utils.py::ReadCSVToDataframeTests::test_non_utf8_encoding
FAILED tests/test_validation.py::CSVValidation::test_csv_not_utf_8 - TypeErro...
```

Looking at Pandas 1.2.1 change notes they list:
> Fixed regression in read_csv() and other read functions were the encoding error policy (errors) did not default to "replace" when no encoding was specified (GH38989)

Output of the returned `pandas.df` (that didn't error because of now replacing) from `test_non_utf8_encoding`
```python
  header  Unnamed: 1
0    f��         NaN
1    bar         NaN
```
I can't remember how the discussion went about if we liked this replacing approach or didn't...

--- a/src/natcap/invest/utils.py
+++ b/src/natcap/invest/utils.py
@@ -582,13 +582,9 @@ def read_csv_to_dataframe(
     # allow encoding kwarg to override this if it's provided
     if not encoding and has_utf8_bom(path):
         encoding = 'utf-8-sig'
-    try:
-        dataframe = pandas.read_csv(
-            path, engine=engine, encoding=encoding, sep=sep, **kwargs)
-    except UnicodeDecodeError as error:
-        LOGGER.error(
-            f'{path} must be encoded as utf-8 or avoid non-ASCII characters')
-        raise error
+    dataframe = pandas.read_csv(path, engine=engine, encoding=encoding,
+                                sep=sep, **kwargs)
+
     # this won't work on integer types, which happens if you set header=None
     # however, there's little reason to use this function if there's no header
     dataframe.columns = dataframe.columns.str.strip()

--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -871,7 +871,7 @@ class ReadCSVToDataframeTests(unittest.TestCase):
         self.assertEqual(df['HEADER2'][1], 5)
 
     def test_non_utf8_encoding(self):
-        """utils: test non-ASCII chars with non-UTF8 encoding raises error"""
+        """utils: test that non-UTF8 encoding doesn't raise an error"""
         from natcap.invest import utils
 
         csv_file = os.path.join(self.workspace_dir, 'csv.csv')
@@ -885,12 +885,13 @@ class ReadCSVToDataframeTests(unittest.TestCase):
                 bar
                 """
             ).strip())
-        with self.assertRaises(UnicodeDecodeError) as cm:
-            # In pandas 1.2.0, there's a bug where sep=None combined
-            # with this UnicodeDecodeError leaves the file handle open,
-            # and our tearDown function errors. So setting sep=',' for now.
-            utils.read_csv_to_dataframe(csv_file, sep=',')
-        self.assertTrue("decode byte" in str(cm.exception))
+
+        df = utils.read_csv_to_dataframe(csv_file)
+        # the default engine='python' should replace the unknown characters
+        # different encodings of replacement character depending on the system
+        self.assertTrue(df['header'][0] in ['f\xce\xce', 
+            'f\N{REPLACEMENT CHARACTER}\N{REPLACEMENT CHARACTER}'])
+        self.assertEqual(df['header'][1], 'bar')
 
     def test_override_default_encoding(self):
         """utils: test that you can override the default encoding kwarg"""

--- a/tests/test_validation.py
+++ b/tests/test_validation.py
@@ -650,7 +650,7 @@ class CSVValidation(unittest.TestCase):
         self.assertTrue('missing from this table' in error_msg)
 
     def test_csv_not_utf_8(self):
-        """Validation: test non-UTF8 CSVs w/ non-ASCII chars return msg."""
+        """Validation: test that non-UTF8 CSVs can validate."""
         from natcap.invest import validation
 
         df = pandas.DataFrame([
@@ -664,8 +664,12 @@ class CSVValidation(unittest.TestCase):
         # https://en.wikipedia.org/wiki/ISO/IEC_8859-5
         df.to_csv(target_file, encoding='iso8859_5')
 
+        # Note that non-UTF8 encodings should pass this check, but aren't
+        # actually being read correctly. Characters outside the ASCII set may
+        # be replaced with a replacement character.
+        # UTF16, UTF32, etc. will still raise an error.
         error_msg = validation.check_csv(target_file)
-        self.assertTrue('File must be encoded as a UTF-8 CSV' in error_msg)
+        self.assertEqual(error_msg, None)
 
     def test_excel_missing_fieldnames(self):
         """Validation: test that we can check missing fieldnames in excel."""

